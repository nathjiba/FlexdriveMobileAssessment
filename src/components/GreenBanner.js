import React from 'react';
import {TouchableOpacity, Image, Text, View} from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';
import closeIcon from '../assets/close_icon.png'

const GreenBanner = props => {
    return (
        <View style={[styles.mainContainer, props.style]}>
            <Text style={styles.bannerTextStyle}>{props.text}</Text>
            <TouchableOpacity onPress={props.closeBanner}>
                <Image source={closeIcon}></Image>
            </TouchableOpacity>
        </View>
    );
}

const styles = EStyleSheet.create({
    mainContainer: {
        padding: 10,
        backgroundColor: '$appGreen',
        alignItems: 'center',
        flexDirection: 'row',
        justifyContent: 'space-between'
    },
    bannerTextStyle: {
        fontSize: 12.5,
        color: 'white',
        marginRight: 5
    }
});

export default GreenBanner;