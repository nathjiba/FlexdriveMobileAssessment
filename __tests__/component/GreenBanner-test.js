import React from 'react';
import GreenBanner from '../../src/components/GreenBanner';
import renderer from 'react-test-renderer';

test('should render without crash', () => {
    const tree = renderer
        .create(<GreenBanner text={'The price includes insurance, maintenance & Road side.'}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});