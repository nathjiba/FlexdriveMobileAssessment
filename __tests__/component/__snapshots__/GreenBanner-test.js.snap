// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crash 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <Text>
    The price includes insurance, maintenance & Road side.
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/close_icon.png",
        }
      }
    />
  </View>
</View>
`;
